lib/auth/init.go

```diff
@@ -342,6 +324,30 @@ func Init(cfg InitConfig, seedConfig bool) (*AuthServer, *Identity, error) {
 	return asrv, identity, nil
 }
 
+func migrateCertAuthority(asrv *AuthServer, in services.CertAuthority) error {
+	raw, ok := (in.GetRawObject()).(services.CertAuthorityV1)
+	if !ok {
+		return nil
+	}
+	_, err := asrv.GetRole(services.RoleNameForCertAuthority(in.GetClusterName()))
+	if err == nil {
+		return nil
+	}
+	if !trace.IsNotFound(err) {
```

Comment:
This (not found) error would have to come from the storage back-end? Do we enforce the proper error type in the tests? Historically there have been bugs in back-ends returning inconsistent error types. I will check, but meanwhile I wonder if we can skip the checking for a specific error type here. Possible?